controllers/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -312,8 +312,11 @@ func (r *Reconciler) deleteUserAccount(logger logr.Logger, targetCluster, name s
 		}
 		return requeueTime, nil
 	}
-
-	if err := memberCluster.Client.Delete(context.TODO(), userAcc); err != nil {
+	propagationPolicy := v1.DeletePropagationForeground
+	err = memberCluster.Client.Delete(context.TODO(), userAcc, &client.DeleteOptions{
+		PropagationPolicy: &propagationPolicy,
```

Comment:
I don't think this will work. There is no owner reference between UA and MUR. They are in different clusters/namespaces.

Reply:
@alexeykazakov But there is an owner reference between UA and NSTemplateSet, and they are in the same cluster.

Reply:
Right.. so this will take care of foreground deletion between UA and dependent objects like NSTemplateSet.. Not between MUR and UA.
So, please disregard my comment :)